imports/client/components/CallSection.tsx

```diff
@@ -326,28 +320,33 @@ const PeerBox = ({
     };
   }, [peer.createdBy]);
 
-  // This stream will have tracks added to/removed from it by the
-  // ConsumerManager children components
   const { current: stream } = useRef(new MediaStream());
-  const [streamActive, setStreamActive] = useState(false);
-  useEffect(() => {
-    const updateActive = (evt: Event) => setStreamActive(evt.type === 'active');
-    stream.addEventListener('active', updateActive);
-    stream.addEventListener('inactive', updateActive);
-    return () => {
-      stream.removeEventListener('active', updateActive);
-      stream.removeEventListener('inactive', updateActive);
-    };
+
+  const [tracks, setTracks] = useState<Map<string, MediaStreamTrack>>(new Map());
+  const setTrack = useCallback((consumer: string, track?: MediaStreamTrack) => {
+    setTracks((prevTracks) => {
+      const prevTrack = prevTracks.get(consumer);
+      const newTracks = new Map(prevTracks);
+
+      if (prevTrack) {
+        stream.removeTrack(prevTrack);
+        newTracks.delete(consumer);
+      }
+      if (track) {
+        stream.addTrack(track);
+        newTracks.set(consumer, track);
+      }
+
+      return newTracks;
+    });
   }, [stream]);
 
-  const audioRef = useCallback((audio: HTMLAudioElement | null) => {
-    if (!streamActive || !audio) {
-      return;
+  const audioRef = useRef<HTMLAudioElement>(null);
+  useEffect(() => {
+    if (audioRef.current && tracks.size > 0) {
+      audioRef.current.srcObject = stream;
     }
-
-    // eslint-disable-next-line no-param-reassign
-    audio.srcObject = stream;
-  }, [streamActive, stream]);
+  }, [audioRef, tracks, stream]);
```

Comment:
ref objects never change, so you don't need to list audioRef in the deps here

Reply:
fixed